### PR TITLE
[Test][Cephadm] Add test to validate host add without NOPASSWD in sud…

### PR DESCRIPTION
**Pull request by pranavprakash20**
…oers

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Comment by pranavprakash20** [comment]
Run log: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-VUJ2HM/

File: suites/reef/cephadm/tier4-service-operations.yaml [hunk 1]
```diff
@@ -171,3 +171,9 @@ tests:
       desc: Validate the key rotation feature
       polarion-id: CEPH-83593829
       module: test_validate_key_rotate.py
+
+  - test:
```
Comment (MohitBis) [inline suites/reef/cephadm/tier4-service-operations.yaml:175]: Hi @pranavprakash20 

Only applicable to reef?

Can we add same test to squid as well?

Comment (pranavprakash20) [reply suites/reef/cephadm/tier4-service-operations.yaml:175]: @MohitBis Noted. Added

**Review by MohitBis — DISMISSED** [review]
LGTM

File: tests/cephadm/test_add_host_with_nopasswd_sudoers_file.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    node.exec_command(cmd=cmd, sudo=True)
+
+    # Verify the update was successful
+    out, _ = node.exec_command(cmd="cat /etc/sudoers.d/*", sudo=True)
+    if file_content not in out:
+        raise OperationFailedError("Failed to update sudoers.d file")
+
+
+def run(ceph_cluster, **kw):
+    """Verify host add loalhost with NOPASSWD removed in sudoers file
```
Comment (vinpapnoi) [inline tests/cephadm/test_add_host_with_nopasswd_sudoers_file.py:22]: nit: localhost

File: tests/cephadm/test_add_host_with_nopasswd_sudoers_file.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        content = "cephuser ALL=(ALL) ALL"
+        update_sudoers_file(node, content)
+
+        # Now verify the host (local host) addition
+        out = CephAdm(node).ceph.orch.host.add(
+            hostname=node.hostname, ip_address=node.ip_address, label="_admin"
+        )
+        if "Can't communicate with remote host " in out:
+            raise OperationFailedError(
+                "Failed to add host(loalhost) with NOPASSWD removed in sudoers file. "
```
Comment (vinpapnoi) [inline tests/cephadm/test_add_host_with_nopasswd_sudoers_file.py:38]: nit: localhost

File: tests/cephadm/test_add_host_with_nopasswd_sudoers_file.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+                "Possible regression of #2115462"
+            )
+    except Exception as e:
+        raise OperationFailedError(
+            f"Faiure while validating the host addition with NOPASSWD removed in sudoers file. {e}"
```
Comment (vinpapnoi) [inline tests/cephadm/test_add_host_with_nopasswd_sudoers_file.py:43]: nit: "failure"

File: cli/ceph/orch/host.py [hunk 5]
```diff
@@ -43,3 +43,17 @@ def maintenance(self, hostname, operation, force=False, yes_i_really_mean_it=Fal
         if not out:
             return False
         return True
+
+    def add(self, hostname, ip_address, label):
```
Comment (vinpapnoi) [inline cli/ceph/orch/host.py:47]: Can we keep "label" as optional here?
```
if label:
    cmd += {label}
```

something like this ?

**Review by vinpapnoi — APPROVED** [review]
LGTM

**Review by sayaleeraut — APPROVED** [review]
LGTM!

**Review by MohitBis — APPROVED** [review]
LGTM

**Review by vamahaja — APPROVED** [review]

**Pull request merged**